### PR TITLE
[Driver] Arbitrary precision integers are only for FPGA and emulation

### PR DESCRIPTION
**Pull request by LU-JOHN**
Only add SPV_INTEL_arbitrary_precision_integers for FPGA and emulation targets.

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -10383,6 +10382,7 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
     if (C.getDriver().IsFPGAHWMode())
       // Enable several extensions on FPGA H/W exclusively
       ExtArg += ",+SPV_INTEL_usm_storage_classes,+SPV_INTEL_runtime_aligned"
+                ",+SPV_INTEL_arbitrary_precision_integers"
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/Clang.cpp:10385]: The description states `FPGA` and does not scrutinize between hardware or emulation.  Should this also apply to emulation?

Comment (LU-JOHN) [reply clang/lib/Driver/ToolChains/Clang.cpp:10385]: @MrSidims Should this also apply to emulation?

Comment (MrSidims) [reply clang/lib/Driver/ToolChains/Clang.cpp:10385]: Yes, the extension should be enabled for both hardware and emulation in AOT compilation.

Comment (LU-JOHN) [reply clang/lib/Driver/ToolChains/Clang.cpp]: Updated to add SPV_INTEL_arbitrary_precision_integers for FPGA emulation targets.

**Review by mdtoguchi — APPROVED** [review]

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 2]
```diff
@@ -10394,6 +10393,9 @@ static void getOtherSPIRVTransOpts(Compilation &C,
               ",+SPV_INTEL_tensor_float32_conversion"
               ",+SPV_INTEL_optnone"
               ",+SPV_KHR_non_semantic_info";
+  if (C.getDriver().IsFPGAHWMode() || C.getDriver().IsFPGAEmulationMode())
```
Comment (MrSidims) [inline clang/lib/Driver/ToolChains/Clang.cpp:10396]: @mdtoguchi was IsFPGAEmulationMode function fixed? I remember it was doing funny things, basically it returns true by default even for JIT.

Comment (MrSidims) [reply clang/lib/Driver/ToolChains/Clang.cpp:10396]: Nvm, I see https://github.com/intel/llvm/issues/4061 is marked as resolved.

**Review by MrSidims — APPROVED** [review]

**Pull request merged**